packaging/flavorgen/flavors/generators.go

```diff
@@ -331,6 +341,105 @@ func defaultPreKubeadmCommands() []string {
 	}
 }
 
+func kubeVIPPod() string {
+	hostPathType := v1.HostPathFileOrCreate
+	pod := &v1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       typeToKind(&v1.Pod{}),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "kube-vip",
+			Namespace: "kube-system",
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:  "kube-vip",
+					Image: "plndr/kube-vip:0.1.6",
+					Args: []string{
+						"start",
+					},
+					ImagePullPolicy: v1.PullIfNotPresent,
+					SecurityContext: &v1.SecurityContext{
+						Capabilities: &v1.Capabilities{
+							Add: []v1.Capability{
+								"NET_ADMIN",
+								"SYS_TIME",
+							},
+						},
+					},
+					VolumeMounts: []v1.VolumeMount{
+						{
+							MountPath: "/etc/kubernetes/admin.conf",
+							Name:      "kubeconfig",
+						},
+					},
+					Env: []v1.EnvVar{
+						{
+							Name:  "vip_arp",
+							Value: "true",
+						},
+						{
+							Name:  "vip_leaderelection",
+							Value: "true",
+						},
+						{
+							Name:  "vip_address",
+							Value: controlPlaneEndpointVar,
+						},
+						{
+							Name:  "lb_backendport",
+							Value: "6443",
+						},
+						{
+							Name:  "vip_addpeerstolb",
+							Value: "true",
+						},
+						{
+							Name:  "lb_name",
+							Value: "kcpEdnpoint",
+						},
+						{
+							Name:  "lb_bindtovip",
+							Value: "true",
+						},
+						{
+							Name:  "vip_interface",
+							Value: vipNetworkInterfaceVar,
+						},
+						{
+							Name:  "lb_type",
+							Value: "tcp",
+						},
+						{
+							Name:  "lb_type",
+							Value: "tcp",
+						},
+					},
+				},
+			},
+			HostNetwork: true,
+			Volumes: []v1.Volume{
+				{
+					Name: "kubeconfig",
+					VolumeSource: v1.VolumeSource{
+						HostPath: &v1.HostPathVolumeSource{
+							Path: "/etc/kubernetes/admin.conf",
+							Type: &hostPathType,
```

Comment:
Let's use `pointer.StringPtr()`

Reply:
`pointer.StringPtr()` would return a `*string`, here we need a `*HostPathType, I'm not sure we can cast this ? did you have something specific in mind ?

Reply:
Sorry, assumed it was a *string.